casper/src/main/scala/coop/rchain/casper/BlockCreator.scala

```diff
@@ -65,26 +78,24 @@ object BlockCreator {
     spanF.trace(CreateBlockMetricsSource) {
       import cats.instances.list._
       for {
-        tipHashes             <- Estimator[F].tips(dag, genesis)
-        _                     <- spanF.mark("after-estimator")
-        parentMetadatas       <- EstimatorHelper.chooseNonConflicting(tipHashes, dag)
+        tipHashes <- Estimator[F].tips(dag, genesis)
+        _ <- Log[F].info(
+              s"Creating block with tip hashes ${tipHashes.map(PrettyPrinter.buildString)}"
+            )
+        _               <- spanF.mark("after-estimator")
+        parentMetadatas <- EstimatorHelper.chooseNonConflicting(tipHashes, dag)
+        _ <- Log[F].info(
+              s"Creating block with parents ${parentMetadatas.map(b => PrettyPrinter.buildString(b.blockHash))}"
+            )
         maxBlockNumber        = ProtoUtil.maxBlockNumberMetadata(parentMetadatas)
+        _                     <- Log[F].info(s"Creating block with maxBlockNumber ${maxBlockNumber}")
         invalidLatestMessages <- ProtoUtil.invalidLatestMessages(dag)
         deploys               <- extractDeploys(dag, parentMetadatas, maxBlockNumber, expirationThreshold)
         sender                = ByteString.copyFrom(validatorIdentity.publicKey.bytes)
         latestMessageOpt      <- dag.latestMessage(sender)
         seqNum                = latestMessageOpt.fold(0)(_.seqNum) + 1
-        // TODO: Add `slashingDeploys` to DeployStorage
-        slashingDeploys = invalidLatestMessages.values.toList.map(
-          invalidBlockHash =>
-            // TODO: Do something useful with the result of "slash".
-            SlashDeploy(
-              invalidBlockHash,
-              validatorIdentity.publicKey,
-              SystemDeployUtil.generateSlashDeployRandomSeed(sender, seqNum)
-            )
-        )
-        parents <- parentMetadatas.toList.traverse(p => ProtoUtil.getBlock(p.blockHash))
+        _                     <- Log[F].info(s"Creating block with seqNum ${seqNum}")
+        parents               <- parentMetadatas.toList.traverse(p => ProtoUtil.getBlock(p.blockHash))
```

Comment:
I didn't look in detail but it might be that we don't need the full block here and just use block metadata instead. :thinking:

Reply:
Ok, we will need at least bonds list which is not stored in bock metadata. Something for later refactoring with the DAG storage indexes.

Reply:
`BlockMetaData` doesn't contain `postStateHash`

Reply:
Maybe we should reconsider `blockMetaData` when we do the migrations.